### PR TITLE
 [minor] [MASCORE-3232] Setup ingress-controller for cis-proxy #1141 

### PR DESCRIPTION
**Pull request by padmankosalaram**
Issue https://jsw.ibm.com/browse/MASCORE-3232

As part of this changes below are achieved:
1. Gitops creates a public IngressController controlled via ingress flag present in cluster params file
2. Gitops adds type: "external" label to mas core  & apps controlled via ingress flag
3. Gitops adds type: "external" label to db2 route controlled via jdbc_route flag, if set to public in cluster params file

### CODE REVIEW

File: instance-applications/130-ibm-mas-suite/templates/05-postsync-add-label_Job.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  labels:
+{{ .Values.custom_labels | toYaml | indent 4 }}
+{{- end }}
+subjects:
+  - kind: ServiceAccount
+    name: mas-route-sa
+    namespace: mas-{{ .Values.instance_id }}-core
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
```
Comment (whitfiea) [inline instance-applications/130-ibm-mas-suite/templates/05-postsync-add-label_Job.yaml:59]: I don't think this needs clusteradmin permission as it is only patching a route in the core namespace

File: instance-applications/510-550-ibm-mas-suite-app-config/templates/04-postsync-add-label_Job.yaml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  labels:
+{{ .Values.custom_labels | toYaml | indent 4 }}
+{{- end }}
+subjects:
+  - kind: ServiceAccount
+    name: mas-app-route-sa
+    namespace: {{ $ns }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
```
Comment (whitfiea) [inline instance-applications/510-550-ibm-mas-suite-app-config/templates/04-postsync-add-label_Job.yaml:62]: Although this needs to be a cluster role, it only has to be a role that can patch a route and not full clusteradmin

**Review by whitfiea — CHANGES_REQUESTED** [review]

File: instance-applications/130-ibm-mas-suite/templates/05-postsync-add-label_Job.yaml (hunk 1 above)
Comment (padmankosalaram) [reply instance-applications/130-ibm-mas-suite/templates/05-postsync-add-label_Job.yaml:98]: added specific roles to list the suite and patch the role

File: instance-applications/510-550-ibm-mas-suite-app-config/templates/04-postsync-add-label_Job.yaml (hunk 2 above)
Comment (padmankosalaram) [reply instance-applications/510-550-ibm-mas-suite-app-config/templates/04-postsync-add-label_Job.yaml:116]: added specific roles to list the app workspaces and patch the routes

**Comment by padmankosalaram** [comment]
closing this PR as its tracked against https://github.com/ibm-mas/gitops/pull/181

**Pull request closed without merging**